src/algebra/star/basic.lean

```diff
@@ -211,29 +211,45 @@ instance star_ring.to_star_add_monoid [semiring R] [star_ring R] : star_add_mono
 def star_ring_equiv [semiring R] [star_ring R] : R ≃+* Rᵐᵒᵖ :=
 { to_fun := λ x, mul_opposite.op (star x),
   ..star_add_equiv.trans (mul_opposite.op_add_equiv : R ≃+ Rᵐᵒᵖ),
-  ..star_mul_equiv}
+  ..star_mul_equiv }
 
-/-- `star` as a `ring_aut` for commutative `R`. This is used to denote complex
-conjugation, and is available under the notation `conj` in the locale `complex_conjugate` -/
+/-- `star` as a ring automorphism, for commutative `R`. -/
+@[simps apply]
 def star_ring_aut [comm_semiring R] [star_ring R] : ring_aut R :=
 { to_fun := star,
   ..star_add_equiv,
   ..star_mul_aut }
 
-localized "notation `conj` := star_ring_aut" in complex_conjugate
+variables (R)
+/-- `star` as a ring endomorphism, for commutative `R`. This is used to denote complex
+conjugation, and is available under the notation `conj` in the locale `complex_conjugate`.
+
+Note that this is the preferred form (over `star_ring_aut`, available under the same hypotheses)
+because the notation `E →ₗ⋆[R] F` for an `R`-conjugate-linear map (short for
+`E →ₛₗ[star_ring_end R] F`) does not pretty-print if there is a coercion involved, as would be the
+case for `(↑star_ring_aut : R →* R)`. -/
+def star_ring_end [comm_semiring R] [star_ring R] : R →+* R :=
+{ to_fun := star,
+  map_one' := star_one R,
+  map_mul' := star_mul',
+  map_zero' := star_zero R,
+  map_add' := star_add }
+variables {R}
+
+localized "notation `conj` := star_ring_end _" in complex_conjugate
 
-/-- This is not a simp lemma, since we usually want simp to keep `star_ring_aut` bundled.
+/-- This is not a simp lemma, since we usually want simp to keep `star_ring_end` bundled.
  For example, for complex conjugation, we don't want simp to turn `conj x`
  into the bare function `star x` automatically since most lemmas are about `conj x`. -/
-lemma star_ring_aut_apply [comm_semiring R] [star_ring R] {x : R} :
-  star_ring_aut x = star x := rfl
+lemma star_ring_end_apply [comm_semiring R] [star_ring R] {x : R} :
```

Comment:
Can `simps` generate this as well?

Reply:
The issue is that we don't want it to be a simp-lemma, see the docstring above.  Is it possible for `simps` to generate a lemma but not actually make it `@[simp]`?

Reply:
Yes, I think  `@[simps {attrs := []}]` does it

Reply:
And then you can use `add_decl_doc` to attach the docstring to it